src/Cli/ConfigGenerator.cs

```diff
@@ -1199,6 +1218,74 @@ private static bool TryUpdateConfiguredAzureLogAnalyticsOptions(
             }
         }
 
+        /// <summary>
+        /// Updates the file sink options in the configuration.
+        /// </summary>
+        /// <param name="options">The configuration options provided by the user.</param>
+        /// <param name="fileOptions">The file sink options to be updated.</param>
+        /// <returns>True if the options were successfully updated; otherwise, false.</returns>
+        private static bool TryUpdateConfiguredFileOptions(
+            ConfigureOptions options,
+            ref FileSinkOptions fileOptions)
+        {
+            try
+            {
+                // Runtime.Telemetry.File.Enabled
+                if (options.FileSinkEnabled is not null)
+                {
+                    fileOptions = fileOptions with { Enabled = options.FileSinkEnabled is CliBool.True, UserProvidedEnabled = true };
+                    _logger.LogInformation($"Updated configuration with runtime.telemetry.file.enabled as '{options.FileSinkEnabled}'");
+                }
+
+                // Runtime.Telemetry.File.Path
+                if (options.FileSinkPath is not null)
+                {
+                    fileOptions = fileOptions with { Path = options.FileSinkPath, UserProvidedPath = true };
+                    _logger.LogInformation($"Updated configuration with runtime.telemetry.file.path as '{options.FileSinkPath}'");
+                }
+
+                // Runtime.Telemetry.File.RollingInterval
+                if (options.FileSinkRollingInterval is not null)
+                {
+                    fileOptions = fileOptions with { RollingInterval = options.FileSinkRollingInterval, UserProvidedRollingInterval = true };
```

Comment:
Will this give error if the rolling interval specified by the user is not one of the expected enums?